src/runtime/tracing.cpp

```diff
@@ -149,9 +149,12 @@ WEAK int32_t default_trace(void *user_context, const halide_trace_event *e) {
                         ss << ((uint64_t *)(e->value))[i];
                     }
                 } else if (e->type_code == 2) {
-                    halide_assert(user_context, print_bits >= 32 && "Tracing a bad type");
+                    halide_assert(user_context, print_bits >= 16 && "Tracing a bad type");
                     if (print_bits == 32) {
                         ss << ((float *)(e->value))[i];
+                    } else if (print_bits == 16) {
+                        // TODO: We should also emit a decimal approximation
+                        ss.write_float16_from_bits_as_hex_float( ((uint16_t *)(e->value))[i] );
```

Comment:
The second option sounds best to me. With a typedef there's a risk of operators not doing what they look like they do.